include/core/SurfaceLayer.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef __NGEN_SURFACE_LAYER__
+#define __NGEN_SURFACE_LAYER__
+
+#include "Layer.hpp"
+
+namespace ngen
+{
+    class SurfaceLayer : public Layer
+    {
+        public:
+        
+        SurfaceLayer(
+                const LayerDescription& desc, 
+                const std::vector<std::string>& p_u, 
+                const Simulation_Time& s_t, 
+                feature_type& f, 
+                geojson::GeoJSON cd, 
+                std::shared_ptr<pdm03_struct> pdmd,
+                long idx,
+                const std::vector<std::string>& n_u,
+                std::unordered_map<std::string, std::ofstream>& output_files) : 
+                    Layer(desc,p_u,s_t,f,cd,pdmd,idx), 
+                    nexus_ids(n_u), 
+                    nexus_outfiles(output_files)
+        {
+
+        }
+
+        /***
+         * @breif Run one simulation timestep for each model in this layer
+        */
+        virtual void update_models()
```

Comment:
The function implementations belong in a separate .cpp file. We shouldn't blow up ngen.cpp compilation time by having them here. It won't even help optimization, because virtual dispatch will impede inlining or other inter-procedural analysis/transformation anyway.

Reply:
This has been done